packages/js/src/plugins/candyMachineModule/operations/findCandyMachinesByPublicKeyField.ts

```diff
@@ -121,8 +122,45 @@ export const findCandyMachinesByPublicKeyFieldOperationHandler: OperationHandler
             ? parseCandyMachineCollectionAccount(unparsedCollectionAccount)
             : null;
 
-          return toCandyMachine(account, unparsedAccount, collectionAccount);
+          return {
+            account,
+            unparsedAccount,
+            collectionAccount,
+          };
         }
       );
+
+      // Find mint details for all unique SPL tokens used
+      // in candy machines that have non-null `tokenMint`
+      const tokenMints = new Set(
```

Comment:
Just curious: Does `Set` works with the `PublicKey` class? How does it know how to identify duplicate values? Is it doing it by reference or is there a magic method that `Set` picks up on?

Reply:
I intended to map to string there - good catch.